### PR TITLE
No Cupy install

### PR DESCRIPTION
**Pull request by Fhrozen**
Adding cupy label to makefile and no-cupy flag
Referred to #674.
Did some modifications in the makefile to reuse the CUPY_VERSION flag.
in case empty, the cupy installation is avoided.
I tested with both `make` and `make CUPY_VERSION=''` without any problem.
Let me know if the modifications are acceptable to modify the Readme file also.

### CODE REVIEW

File: tools/check_install.py [hunk 1]
```diff
@@ -8,110 +8,123 @@
...
-        if hasattr(lib, "__version__"):
-            is_correct = lib.__version__ in version
-            if is_correct:
-                logging.info("--> %s version is matched." % name)
-                is_correct_version_list.append(True)
+
+def main(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--no-cupy', action='store_true', default=False,
+                        help='Number of GPUs')
```
Comment (kan-bayashi) [inline tools/check_install.py]: Fix help comment.

File: tools/Makefile [hunk 2]
```diff
@@ -62,10 +62,18 @@ espnet.done: venv
...
 	touch espnet.done
 endif
 
+ifneq ($(strip $(CUPY_VERSION)),)
+cupy.done: venv
+	. venv/bin/activate && pip install cupy==$(CUPY_VERSION)
+	touch cupy.done
+else
+cupy.done: venv
+	touch cupy.done
```
Comment (kan-bayashi) [inline tools/Makefile]: I’m not sure cupy.done is needed if we do not install cupy.

File: tools/check_install.py (hunk 1 above)
Comment (Fhrozen) [reply tools/check_install.py]: resolved in new commit

File: tools/Makefile (hunk 2 above)
Comment (Fhrozen) [reply tools/Makefile]: resolved in new commit

**Comment by Fhrozen** [comment]
Pushed a new commit resolving observations from @kan-bayashi 
Moved cupy to be installed under the chainer_patch target, if CUPY_VERSION has a value.
Test both installations without problems.
Let me know if there is any possible conflict or wrong part.

**Comment by kan-bayashi** [comment]
Sorry, my intension is as follows:

```
cupy.done: espnet.done
ifneq ($(strip $(CUPY_VERSION)),)
	. venv/bin/activate && pip install cupy==$(CUPY_VERSION)
        touch cupy.done
endif
```

**Comment by sw005320** [comment]
I like @kan-bayashi's idea. @Fhrozen, can you change it like that?

**Comment by Fhrozen** [comment]
Just finished the test for @kan-bayashi's idea. @sw005320 Let me know if there is any additional part to modify.

**Pull request merged**